src/odemis/driver/tescan.py

```diff
@@ -1096,6 +1106,58 @@ def _pollXYZ(self):
         except Exception:
             logging.exception("Unexpected failure during XYZ polling")
 
+    def _checkPosition(self, ref_pos: Dict[str, float], pos: Dict[str, float], rel: bool = False):
+        """
+        Checks if the current stage position is equal to the given position.
+        :param ref_pos: stage position before movement
+        :param pos: requested stage position in relative or absolute coordinates
+        :param rel: By default False, requested stage position is in absolution position otherwise relative
+        """
```

Comment:
Please add a :raise: docstring to indicate it will fail if the stage doesn't reach the target position within a given time.